SearchService2/appscale/search/search_server.py

```diff
@@ -0,0 +1,190 @@
+        remote_api_response.response = encoded_response
+
+      # Write error details to log
+      if err.error_code == search_pb2.SearchServiceError.INTERNAL_ERROR:
+        logger.exception('InternalError: {}'.format(err.error_detail))
+      else:
+        logger.warning('SearchServiceError "{}" ({})'
+                       .format(err.error_name, err.error_detail))
+
+    # Write encoded Remote API response
+    self.write(remote_api_response.SerializeToString())
+
+  @gen.coroutine
+  def handle_search_api_request(self, app_id, search_api_method,
+                                search_api_req_data):
+    """ Handles Search API request.
+
+    Args:
+      app_id: A string representing project_id.
+      search_api_method: A string representing name of Search API method.
+      search_api_req_data: Encoded protobuffer Search API request.
+    Returns:
+      An instance of protobuffer Search API response.
+    """
+    try:
+      req_class, resp_class, executor = self.api[search_api_method]
+    except KeyError:
+      raise SearchServiceError(
+        search_pb2.SearchServiceError.INVALID_REQUEST,
+        'Unknown request method "{}"'.format(search_api_method)
+      )
+    search_api_req = req_class()
+    search_api_req.ParseFromString(search_api_req_data)
+    if not search_api_req.app_id:
+      search_api_req.app_id = app_id
+    search_api_resp = resp_class()
+    yield executor(search_api_req, search_api_resp)
+    raise gen.Return(search_api_resp)
+
+
+def prepare_api_methods(zk_client):
+  """ Instantiates APIMethods and defines API methods routing.
+
+  Args:
+    zk_client: an instance of kazoo.client.KazooClient.
+  Returns:
+    a dict which maps Search API method name to a tuple of
+    (<request_class>, <response_class>, <coroutine_to_handle_request>)
+  """
+  api = api_methods.APIMethods(zk_client)
+  return {
+    'IndexDocument': (
+      search_pb2.IndexDocumentRequest,
+      search_pb2.IndexDocumentResponse,
+      api.index_document
+    ),
+    'DeleteDocument': (
+      search_pb2.DeleteDocumentRequest,
+      search_pb2.DeleteDocumentResponse,
+      api.delete_document
+    ),
+    'ListIndexes': (
+      search_pb2.ListIndexesRequest,
+      search_pb2.ListIndexesResponse,
+      api.list_indexes
+    ),
+    'ListDocuments': (
+      search_pb2.ListDocumentsRequest,
+      search_pb2.ListDocumentsResponse,
+      api.list_documents
+    ),
+    'Search': (
+      search_pb2.SearchRequest,
+      search_pb2.SearchResponse,
+      api.search
+    )
+  }
+
+
+def main():
+  """ Start SearchService2 server. """
+  logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+    '-v', '--verbose', action='store_true',
+    help='Output debug-level logging')
+  parser.add_argument(
+    '-p', '--port', type=int, help='The port to listen on')
+  parser.add_argument(
+    '-z', '--zk-location', default=None,
```

Comment:
For argparse, I believe `nargs='+'` is the standard way of taking multiple values and storing them as a list.